### PR TITLE
Add nosuid Option to /boot.

### PR DESCRIPTION
**Pull request by adelton**
#### Description:

- Add nosuid Option to /boot.

#### Rationale:

- Users should not be able to execute SUID or SGID binaries from boot partitions.

### CODE REVIEW

File: linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    The <tt>nosuid</tt> mount option can be used to prevent
+    execution of setuid programs in <tt>/boot</tt>. The SUID and SGID permissions
+    should not be required in these user data directories.
+    {{{ describe_mount(option="nosuid", part="/boot") }}}
+
+rationale: |-
+    The presence of SUID and SGID executables should be tightly controlled. Users
+    should not be able to execute SUID or SGID binaries from boot partitions.
+
+severity: unknown
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml]: A severity should be set.

File: linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+identifiers:
+    cce@rhel8: 81033-3
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml]: Should a CCE be assigned for RHEL 6 and 7?

File: linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+references:
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml]: Can likely nab the NIST references from other partition option checks.

File: linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml (hunk 2 above)
Comment (shawndwells) [reply linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml]: Also, make sure to commit an updated cce-available file.

File: linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+documentation_complete: true
+
+prodtype: rhel8,fedora,ol8
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml]: What about RHEL 7, maybe RHEL 6 too?

File: linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml (hunk 1 above)
Comment (adelton) [reply linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml]: @yuumasato, @stevegrubb, what should be the severity for these `mount_option_*` rules? We currently have `unknown` in most of them, with just `mount_option_dev_shm_*` having `medium`.

File: linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml (hunk 3 above)
Comment (adelton) [reply linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml]: So `nist: CM-7,MP-2` and `nist-csf: PR.IP-1,PR.PT-2,PR.PT-3`?

File: linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml (hunk 4 above)
Comment (adelton) [reply linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml]: I'm working based on the RHEL 8 kickstart by @stevegrubb. Not sure to what extend this is applicable to older versions. We can always add them later en masse.

File: linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml (hunk 2 above)
Comment (adelton) [reply linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml]: Mea culpa, fixed in 8bd439d5c5c558db5b5ae8a6d418fb8b02bec0d9 (also rebased on master).

**Comment by adelton** [comment]
> Should this also be enabled in profiles?

It should. However, ol8/profiles/ospp.profile currently only has `mount_option_dev_shm_*`, so only small subset of what we are tracking as things that are already done.

@yuumasato, do you want me to add all the current `mount_option_*` rules there now and then keep adding the new ones one by one, or do you want just the rules created and then added to the profile(s) in one final commit?

File: linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml (hunk 1 above)
Comment (yuumasato) [reply linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml]: Honestly, I don't know how the `severity` is set.

**Comment by adelton** [comment]
I've now added `nist: CM-7,MP-2` and `nist-csf: PR.IP-1,PR.PT-2,PR.PT-3` references to make the tests pass, and I've added entry to rhel8/templates/csv/mount_options.csv to get OVAL generated. And rebased on master -> 066db6d90e1b8db4a46d73d787524971955bd8f9.

If the references are OK and the rest looks sane, the only thing missing is the severity value.

File: linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+title: 'Add nosuid Option to /boot'
+
+description: |-
+    The <tt>nosuid</tt> mount option can be used to prevent
+    execution of setuid programs in <tt>/boot</tt>. The SUID and SGID permissions
+    should not be required in these user data directories.
```
Comment (yuumasato) [inline linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml]: boot partition shouldn't contain user data.

**Comment by yuumasato** [comment]
I'm not sure about `ospp` reference though.

File: linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml (hunk 5 above)
Comment (adelton) [reply linux_os/guide/system/permissions/partitions/mount_option_boot_nosuid/rule.yml]: Fixed in 22122d07991cf6d5f74e963c91d12913b9c425f9.

**Comment by adelton** [comment]
> `medium` seems reasonable.

Applied in 22122d07991cf6d5f74e963c91d12913b9c425f9.

**Comment by adelton** [comment]
The `ospp` reference is not for example in linux_os/guide/system/permissions/partitions/mount_option_var_tmp_nosuid/rule.yml either.

We can likely add them en masse when we are adding the rules to the `ospp.profile`s.

**Comment by yuumasato** [comment]
> We can likely add them en masse when we are adding the rules to the `ospp.profile`s.

I agree.

**Pull request merged**